jiant/tasks/lib/mlm_pretokenized.py

```diff
@@ -0,0 +1,92 @@
+from dataclasses import dataclass
+from typing import List
+
+import jiant.utils.python.io as py_io
+from jiant.tasks.core import (
+    Task,
+    TaskTypes,
+    BaseExample,
+)
+from jiant.tasks.utils import ExclusiveSpan
+import jiant.tasks.lib.templates.mlm_premasked as mlm_premasked_template
+
+
+@dataclass
+class Example(BaseExample):
+    guid: str
+    tokenized_text: List
+    # Spans over token indices
+    masked_spans: List[ExclusiveSpan]
+
+    def tokenize(self, tokenizer):
+        # masked_tokens will be regular tokens except with tokenizer.mask_token for masked spans
+        # label_tokens will be tokenizer.pad_token except with the regular tokens for masked spans
+        masked_tokens = []
+        label_tokens = []
+        curr = 0
+        for start, end in self.masked_spans:
+            # Handle text before next mask
+            tokenized_text = self.tokenized_text[curr:start]
+            masked_tokens += tokenized_text
+            label_tokens += [tokenizer.pad_token] * len(tokenized_text)
+
+            # Handle mask
+            tokenized_masked_text = self.tokenized_text[start:end]
+            masked_tokens += [tokenizer.mask_token] * len(tokenized_masked_text)
+            label_tokens += tokenized_masked_text
+            curr = end
+        if curr < len(self.tokenized_text):
+            tokenized_text = self.tokenized_text[curr:]
+            masked_tokens += tokenized_text
+            label_tokens += [tokenizer.pad_token] * len(tokenized_text)
+
+        return TokenizedExample(
+            guid=self.guid, masked_tokens=self.tokenized_text, label_tokens=label_tokens,
+        )
```

Comment:
is this the same as tokenize() below? if so, could be refactored

Reply:
This is the portion where we handle the tokenization wrt masking, so it's different from other tasks.